# **Réalisez une étude de santé publique avec R ou Python**
![Guide pratique](https://www.fao.org/images/corporatelibraries/fao-logo/fao-logo-en.svg?sfvrsn=f64522b4_31)

**OBJECTIFS:**

Donner un panorama de l'état de la malnutrition dans le monde, particulièrement pour l'année 2017.

La proportion de personnes en état de sous-nutrition ;

Le nombre théorique de personnes qui pourraient être nourries ;

La disponibilité alimentaire des produits végétaux ;

l’utilisation de la disponibilité intérieure, l’alimentation animale, l'alimentation humaine.

**INTRODUCTION:**

A partir des données provenant du site web officiel de la FAO (Food and Agriculture Organization), nous allons etudier l'état de la malnutrition dans le monde. Pour cela, nous effectuerons les operations suivantes:

- Import et préparation des données
- Analyse des données

**SOMMAIRE:**

**Partie 1 : Préparation des données**

 - <a href="#C3">1.Importer les librairies</a>
 - <a href="#C4">2.Importer le jeu des données</a>
 - <a href="#C5">3.Vérification de chaque dataframe lors de l'importation et pré-traitement des données</a>
 - <a href="#C6">4.Pré_traitement et nettoyage des données</a>

**Partie 2 : Analyse des données**

-  **Demandes de Marc**
- <a href="#C7">1.Proportion de la population en état de sous_nutrition en 2017</a>
- <a href="#C8">2.Nombre théorique de personnes qui pourraient être nourries(à partir de la disponibilité alimentaire mondiale)</a>
- <a href="#C9">3.Nombre théorique de personnes qui pourraient être nourries à partir de la disponibilité alimentaire des produits végéteaux</a>
- <a href="#C10">4.L’utilisation de la disponibilité intérieure</a>
-  **Demandes de Mélanie**
- <a href="#C11">1.Pays pour lesquels la proportion de personnes sous-alimentées est la plus forte en 2017</a>
- <a href="#C12">2.Pays qui ont le plus bénéficié d’aide depuis 2013</a>
- <a href="#C13">3.Pays ayant le plus de disponibilité par habitant en 2017</a>
- <a href="#C14">4.Pays ayant le moins de disponibilité par habitant en 2017</a>
- <a href="#C15">5.Pays qui semblent etre le plus en difficulté,au niveau alimentaire en 2017</a>
-  **Demandes de Julien**
- <a href="#C16">1.Répartition des céréales entre l'alimentation humaine(colonne Nourriture) et l'alimentation pour animaux</a>
- <a href="#C17">2.L'exportation du Manioc par rapport à sa production en thailande</a>


# Partie 1 : Préparation des données

# <a name="C3"><font color='Black'>Importer les librairies</font></a>

In [ ]:
#Pandas est une librairie user friendly qui gère un très grand nombre de données, a beaucoup de fonctionnalités.
#L'avantage est qu'elle soit complementaire avec d'autres librairies. 
#2 structures de données de Pandas : Dataframe(matrice/tableau) / Series
# Numpy est une librairie destineé à manipuler les matrices et les tableaux multidimentionnels.

In [1]:
#Importer les librairies Pandas et Numpy
import pandas as pd
pd.options.mode.chained_assignment = None  # default='warn
# Verifier la version de pandas
print("Pandas verion : " + pd.__version__)
import numpy as np
print("NumPy version : " + np.__version__)

Pandas verion : 1.4.4
NumPy version : 1.21.5


# <a name="C4"><font color='Black'>Importer le jeu des données</font></a>

In [2]:
#Importer les fichier csv à l'aide de la fonction 'read_csv'.
#Les fichiers csv à importer sont:
df_sn=pd.read_csv("sous_nutrition.csv")
df_pop=pd.read_csv("population.csv")
df_da=pd.read_csv("dispo_alimentaire.csv")
df_aa=pd.read_csv("aide_alimentaire.csv")

# <a name="C5"><font color='Black'>Vérification de chaque dataframe lors de l'importation et pré-traitement des données</font></a>

In [3]:
### Les fonctions pour les verifier les données:
#dataframe.describe() 
#dataframe.info()
#dataframe.shape
#dataframe.dtypes
#dataframe.isna().mean()

In [85]:
#Vérifier les différentes informaTions et voir quels variables contiennent des valeurs manquantes
df_sn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Pays            1218 non-null   object 
 1   Année           1218 non-null   int32  
 2   nb_hab_sousnut  624 non-null    float64
dtypes: float64(1), int32(1), object(1)
memory usage: 23.9+ KB


In [86]:
#Vérifier les différentes informaTions et voir quels variables contiennent des valeurs manquantes
df_pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Pays    1416 non-null   object 
 1   Année   1416 non-null   int64  
 2   nb_hab  1416 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 33.3+ KB


In [6]:
#Vérifier les différentes informaTions et voir quels variables contiennent des valeurs manquantes
df_da.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Zone                                                           15605 non-null  object 
 1   Produit                                                        15605 non-null  object 
 2   Origine                                                        15605 non-null  object 
 3   Aliments pour animaux                                          2720 non-null   float64
 4   Autres Utilisations                                            5496 non-null   float64
 5   Disponibilité alimentaire (Kcal/personne/jour)                 14241 non-null  float64
 6   Disponibilité alimentaire en quantité (kg/personne/an)         14015 non-null  float64
 7   Disponibilité de matière grasse en quantité (g/personne/jo

In [7]:
#Vérifier les différentes informaTions et voir quels variables contiennent des valeurs manquantes
df_aa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1475 entries, 0 to 1474
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Pays bénéficiaire  1475 non-null   object
 1   Année              1475 non-null   int64 
 2   Produit            1475 non-null   object
 3   Valeur             1475 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 46.2+ KB


# <a name="C6"><font color='Black'>Pré_traitement et nettoyage des données</font></a>

In [8]:
#Les informations sont demandées pour l’année 2017

Nos jeux de données contiennent :

- (1) Le nombre de population dans chaque pays pour une période donnée.
- (2) Le nombre de personnes en sous_nutrition par pays et par période.
- (3) Les pays bénéficiaires de l'aide alimentaire, le produit et sa quantité.
- (4) La disponibilité alimentaire pour l'année 2017 par pays, le produit, la disponibilité de la composition de ce dernier en 4 variables, 10 variables utiles pour connaitre le moyen d'aquisition de la disponibilité alimentaire et l'origine du produit (animale ou végétale).

# Nettoyage du tableau sous_nutrition

In [9]:
#La première étape après toute importation est de regarder les 5 premières lignes:
# head permet de voir les premiers élements du dataframe
df_sn.head()

Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

In [10]:
# La deuxieme étape est de regarder les différentes informaTions et voir quels variables contiennent des valeurs manquantes afin de les remplacer par la suite
df_sn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Zone    1218 non-null   object
 1   Année   1218 non-null   object
 2   Valeur  624 non-null    object
dtypes: object(3)
memory usage: 28.7+ KB


In [11]:
# Verification s'il y a des valeurs manquantes. 
df_sn.isna().mean()

Zone      0.000000
Année     0.000000
Valeur    0.487685
dtype: float64

Il y a des valeurs manquantes dans une colonne 'Valeur' : 624 lignes au lieu de 1218 lignes.

In [12]:
# Ensuite je renomme la colonne Valeur par nb_hab_sousnut pour avoir un sens plus explicite et faire la différence avec les colonnes des autres tables
df_sn=df_sn.rename(columns={'Valeur':'nb_hab_sousnut','Zone':'Pays'})
df_sn

Pays      Année nb_hab_sousnut
0     Afghanistan  2012-2014            8.6
1     Afghanistan  2013-2015            8.8
2     Afghanistan  2014-2016            8.9
3     Afghanistan  2015-2017            9.7
4     Afghanistan  2016-2018           10.5
...           ...        ...            ...
1213     Zimbabwe  2013-2015            NaN
1214     Zimbabwe  2014-2016            NaN
1215     Zimbabwe  2015-2017            NaN
1216     Zimbabwe  2016-2018            NaN
1217     Zimbabwe  2017-2019            NaN

[1218 rows x 3 columns]

Traiter les valeurs manquantes pour le fichier 'sous_nutrition'

In [13]:
#Je cherche les valeurs en doubles
df_sn.duplicated().sum()

0

In [14]:
#je remplace la valeur '<0.1' (inferieure à 100000 personnes) par '0.01'(impact faible)
df_sn['nb_hab_sousnut']=df_sn['nb_hab_sousnut'].replace(['<0.1'],'0.01')
                                               
df_sn['nb_hab_sousnut']

0        8.6
1        8.8
2        8.9
3        9.7
4       10.5
        ... 
1213     NaN
1214     NaN
1215     NaN
1216     NaN
1217     NaN
Name: nb_hab_sousnut, Length: 1218, dtype: object

In [15]:
#Modification de la colonne année,remplacer la période par une année:
df_sn['Année'].replace(['2012-2014','2013-2015','2014-2016','2015-2017','2016-2018','2017-2019'],['2013','2014','2015','2016','2017','2018'],inplace=True)
df_sn

Pays Année nb_hab_sousnut
0     Afghanistan  2013            8.6
1     Afghanistan  2014            8.8
2     Afghanistan  2015            8.9
3     Afghanistan  2016            9.7
4     Afghanistan  2017           10.5
...           ...   ...            ...
1213     Zimbabwe  2014            NaN
1214     Zimbabwe  2015            NaN
1215     Zimbabwe  2016            NaN
1216     Zimbabwe  2017            NaN
1217     Zimbabwe  2018            NaN

[1218 rows x 3 columns]

In [16]:
#Changement de types des champs 
df_sn=df_sn.astype({'Pays':str,'Année':int,'nb_hab_sousnut':float})
df_sn

Pays  Année  nb_hab_sousnut
0     Afghanistan   2013             8.6
1     Afghanistan   2014             8.8
2     Afghanistan   2015             8.9
3     Afghanistan   2016             9.7
4     Afghanistan   2017            10.5
...           ...    ...             ...
1213     Zimbabwe   2014             NaN
1214     Zimbabwe   2015             NaN
1215     Zimbabwe   2016             NaN
1216     Zimbabwe   2017             NaN
1217     Zimbabwe   2018             NaN

[1218 rows x 3 columns]

In [17]:
#conversion pour avoir les mêmes unités entre les differentes données : *1000000 du nombre d'habitants en sous-nutrition

df_sn['nb_hab_sousnut']=df_sn['nb_hab_sousnut']*1000000
df_sn['nb_hab_sousnut']

0        8600000.0
1        8800000.0
2        8900000.0
3        9700000.0
4       10500000.0
           ...    
1213           NaN
1214           NaN
1215           NaN
1216           NaN
1217           NaN
Name: nb_hab_sousnut, Length: 1218, dtype: float64

In [18]:
#Affiche les lignes ordonnées par odre croissant en fonction du nombre d'habitants en sous nutrition

df_sn.sort_values(['nb_hab_sousnut'])
df_sn

Pays  Année  nb_hab_sousnut
0     Afghanistan   2013       8600000.0
1     Afghanistan   2014       8800000.0
2     Afghanistan   2015       8900000.0
3     Afghanistan   2016       9700000.0
4     Afghanistan   2017      10500000.0
...           ...    ...             ...
1213     Zimbabwe   2014             NaN
1214     Zimbabwe   2015             NaN
1215     Zimbabwe   2016             NaN
1216     Zimbabwe   2017             NaN
1217     Zimbabwe   2018             NaN

[1218 rows x 3 columns]

# Nettoyage du tableau population

In [19]:
#La première étape après toute importation est de regarder les 5 premières lignes du dataframe
    
df_pop.head()

Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [20]:
#La deuxieme étape est de regarder les différentes informaTions et voir quels variables contiennent des valeurs manquantes afin de les remplacer par la suite

df_pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Zone    1416 non-null   object 
 1   Année   1416 non-null   int64  
 2   Valeur  1416 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 33.3+ KB


In [21]:
#Je cherche les valeurs en doubles
df_pop.duplicated().sum()

0

In [22]:
#Ensuite je renomme la colonne Valeur par nb_hab pour avoir un sens plus explicite et faire la différence avec les colonnes des autres tables

df_pop=df_pop.rename(columns={'Valeur':'nb_hab','Zone':'Pays'})
df_pop

Pays  Année     nb_hab
0     Afghanistan   2013  32269.589
1     Afghanistan   2014  33370.794
2     Afghanistan   2015  34413.603
3     Afghanistan   2016  35383.032
4     Afghanistan   2017  36296.113
...           ...    ...        ...
1411     Zimbabwe   2014  13586.707
1412     Zimbabwe   2015  13814.629
1413     Zimbabwe   2016  14030.331
1414     Zimbabwe   2017  14236.595
1415     Zimbabwe   2018  14438.802

[1416 rows x 3 columns]

In [23]:
#changement de types des champs 
df_pop=df_pop.astype({'Pays':str})
df_pop

Pays  Année     nb_hab
0     Afghanistan   2013  32269.589
1     Afghanistan   2014  33370.794
2     Afghanistan   2015  34413.603
3     Afghanistan   2016  35383.032
4     Afghanistan   2017  36296.113
...           ...    ...        ...
1411     Zimbabwe   2014  13586.707
1412     Zimbabwe   2015  13814.629
1413     Zimbabwe   2016  14030.331
1414     Zimbabwe   2017  14236.595
1415     Zimbabwe   2018  14438.802

[1416 rows x 3 columns]

In [24]:
#Conversion pour avoir les mêmes unités entre les differentes données : *1000 du nombre d'habitants

df_pop['nb_hab']=df_pop['nb_hab']*1000
df_pop['nb_hab']


0       32269589.0
1       33370794.0
2       34413603.0
3       35383032.0
4       36296113.0
           ...    
1411    13586707.0
1412    13814629.0
1413    14030331.0
1414    14236595.0
1415    14438802.0
Name: nb_hab, Length: 1416, dtype: float64

In [25]:
#Afficher les lignes ordonnées par odre croissant en fonction du nombre d'habitants

df_pop.sort_values(['nb_hab'])

Pays  Année        nb_hab
1174          Saint-Siège   2017  7.930000e+02
1171          Saint-Siège   2014  8.000000e+02
1170          Saint-Siège   2013  8.010000e+02
1173          Saint-Siège   2016  8.010000e+02
1175          Saint-Siège   2018  8.010000e+02
...                   ...    ...           ...
259   Chine, continentale   2014  1.399454e+09
260   Chine, continentale   2015  1.406848e+09
261   Chine, continentale   2016  1.414049e+09
262   Chine, continentale   2017  1.421022e+09
263   Chine, continentale   2018  1.427648e+09

[1416 rows x 3 columns]

# Nettoyage du tableau dispo_alimentaire

In [26]:
# La première étape après toute importation est de regarder les 5 premières lignes du dataframe
df_da.head()

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

In [27]:
#La deuxième étape est de regarder les différentes informations et voir quels variables contiennent des valeurs manquantes afin de les remplacer par la suite
df_da.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Zone                                                           15605 non-null  object 
 1   Produit                                                        15605 non-null  object 
 2   Origine                                                        15605 non-null  object 
 3   Aliments pour animaux                                          2720 non-null   float64
 4   Autres Utilisations                                            5496 non-null   float64
 5   Disponibilité alimentaire (Kcal/personne/jour)                 14241 non-null  float64
 6   Disponibilité alimentaire en quantité (kg/personne/an)         14015 non-null  float64
 7   Disponibilité de matière grasse en quantité (g/personne/jo

In [28]:
#Je cherche les valeurs en doubles
df_da.duplicated().sum()

0

In [29]:
# Je me renseigne sur les données statistiques des données.
df_da.describe()

Aliments pour animaux  Autres Utilisations  \
count            2720.000000          5496.000000   
mean              479.501838           157.391376   
std              4240.119637          5076.785816   
min                 0.000000             0.000000   
25%                 0.000000             0.000000   
50%                 4.000000             0.000000   
75%                74.000000             4.000000   
max            150000.000000        347309.000000   

       Disponibilité alimentaire (Kcal/personne/jour)  \
count                                    14241.000000   
mean                                        34.789832   
std                                        107.287655   
min                                        -21.000000   
25%                                          0.000000   
50%                                          4.000000   
75%                                         21.000000   
max                                       1711.000000   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                       14015.000000        
mean                                            8.719368        
std                                            24.618223        
min                                            -1.930000        
25%                                             0.060000        
50%                                             0.830000        
75%                                             5.190000        
max                                           430.760000        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                       11794.000000               
mean                                            1.283111               
std                                             3.680399               
min                                            -0.030000               
25%                                             0.010000               
50%                                             0.080000               
75%                                             0.630000               
max                                            60.760000               

       Disponibilité de protéines en quantité (g/personne/jour)  \
count                                       11561.000000          
mean                                            1.223608          
std                                             3.598686          
min                                            -0.370000          
25%                                             0.010000          
50%                                             0.100000          
75%                                             0.660000          
max                                            54.970000          

       Disponibilité intérieure  Exportations - Quantité  \
count              15382.000000             12226.000000   
mean                 640.293460               110.596925   
std                 9067.267153              1053.318990   
min                -3430.000000               -41.000000   
25%                    0.000000                 0.000000   
50%                    7.000000                 0.000000   
75%                   76.750000                 9.000000   
max               739267.000000             42797.000000   

       Importations - Quantité     Nourriture        Pertes     Production  \
count             14852.000000   14015.000000   4278.000000    9180.000000   
mean                 87.264543     347.931359    106.053763    1090.379085   
std                 717.372714    4475.704458   1113.100416   12067.344094   
min                -201.000000    -246.000000      0.000000       0.000000   
25%                   0.000000       0.000000      0.000000       2.000000   
50%                   2.000000       5.000000      4.000000      22.000000   
75%                  18.000000      52.000000     26.000000     191.250000   
max        

In [30]:
#Ensuite je renomme les colonnes suivantes:
df_da=df_da.rename(columns={'Zone':'Pays','Disponibilité alimentaire (Kcal/personne/jour)':'dispo_al',
                   'Disponibilité alimentaire en quantité (kg/personne/an)':'dispo_al_qté',
                   'Disponibilité de matière grasse en quantité (g/personne/jour)':'dispo_mg',
                   'Disponibilité de protéines en quantité (g/personne/jour)':'dispo_prot',
                    'Disponibilité intérieure':'dispo_int'
                            
                  })

In [31]:
#Changement de types des champs 
df_da=df_da.astype({'Pays':str,'Produit':str, 'Origine':str})
df_da.dtypes

Pays                        object
Produit                     object
Origine                     object
Aliments pour animaux      float64
Autres Utilisations        float64
dispo_al                   float64
dispo_al_qté               float64
dispo_mg                   float64
dispo_prot                 float64
dispo_int                  float64
Exportations - Quantité    float64
Importations - Quantité    float64
Nourriture                 float64
Pertes                     float64
Production                 float64
Semences                   float64
Traitement                 float64
Variation de stock         float64
dtype: object

In [32]:
#Conversion pour avoir les mêmes unités entre les differentes données :disponibilite alimentaire en quantité de "kg/personne/an" en "g/personne/jour"
df_da['dispo_al_qté']=df_da['dispo_al_qté']*1000/365
df_da['dispo_al_qté']

0         4.712329
1         3.534247
2         0.164384
3         0.000000
4         7.397260
           ...    
15600    12.876712
15601     9.150685
15602     0.164384
15603     0.191781
15604     1.315068
Name: dispo_al_qté, Length: 15605, dtype: float64

In [33]:
# Afficher les lignes ordonnées par ordre croissant en foction de la dispo_al_qté
df_da.sort_values(['dispo_al_qté'])

Pays                   Produit   Origine  \
7029          Japon                    Avoine  vegetale   
11029       Pologne           Huile de Sésame  vegetale   
4695          Fidji        Huiles de Poissons   animale   
13595         Tchad  Animaux Aquatiques Autre   animale   
1598         Belize        Plantes Aquatiques   animale   
...             ...                       ...       ...   
15560  Îles Salomon        Huile de Tournesol  vegetale   
15569  Îles Salomon              Miscellanees  vegetale   
15576  Îles Salomon                 Palmistes  vegetale   
15581  Îles Salomon  Plantes Oleiferes, Autre  vegetale   
15593  Îles Salomon                    Sorgho  vegetale   

       Aliments pour animaux  Autres Utilisations  dispo_al  dispo_al_qté  \
7029                    50.0                  1.0     -21.0     -5.287671   
11029                    NaN                 50.0       0.0      0.000000   
4695                     0.0                  0.0       0.0      0.000000   
13595                    NaN                  0.0       0.0      0.000000   
1598                     NaN                  0.0       0.0      0.000000   
...                      ...                  ...       ...           ...   
15560                    NaN                  NaN       0.0           NaN   
15569                    NaN                  NaN       3.0           NaN   
15576                    NaN                  5.0       NaN           NaN   
15581                    0.0                  NaN       NaN           NaN   
15593                    0.0                  NaN       NaN           NaN   

       dispo_mg  dispo_prot  dispo_int  Exportations - Quantité  \
7029      -0.03       -0.37     -159.0                    -41.0   
11029      0.00        0.00       50.0                     16.0   
4695       0.00        0.00        0.0                      0.0   
13595      0.00        0.00        0.0                      NaN   
1598       0.00        0.00        0.0                      NaN   
...         ...         ...        ...                      ...   
15560       NaN         NaN        NaN                      NaN   
15569      0.04        0.10        NaN                      NaN   
15576       NaN         NaN        5.0                      8.0   
15581       NaN         NaN        0.0                      0.0   
15593       NaN         NaN        0.0                      NaN   

       Importations - Quantité  Nourriture  Pertes  Production  Semences  \
7029                    -201.0      -246.0     NaN         0.0       0.0   
11029                     61.0         0.0     NaN         5.0       NaN   
4695                       0.0         0.0     NaN         0.0       NaN   
13595                      0.0         0.0     NaN         NaN       NaN   
1598                       0.0         0.0     NaN         0.0       NaN   
...                        ...         ...     ...         ...       ...   
15560                      NaN         NaN     NaN         NaN       NaN   
15569                      NaN         NaN     NaN         NaN       NaN   
15576                      0.0         NaN     NaN        13.0       NaN   
15581                      0.0         NaN     NaN         NaN       NaN   
15593                      0.0         NaN     NaN         NaN       NaN   

       Traitement  Variation de stock  
7029         35.0                 0.0  
11029         NaN                 0.0  
4695          NaN                 NaN  
13595         NaN                 NaN  
1598          NaN                 NaN  
...           ...                 ...  
15560         NaN                 NaN  
15569         NaN                 NaN  
15576         NaN                 0.0  
15581         NaN                 NaN  
15593         NaN                 NaN  

[15605 rows x 18 columns]

# Nettoyage du tableau aide_alimentaire

In [34]:
#La première étape après toute importation est de regarder les 5 premières lignes du dataframe
df_aa.head()

Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

In [35]:
#La deuxième étape est de regarder les différentes informations et de voir quelles variables contiennent des valeurs manquantes afin de les remplacer par la suite
df_aa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1475 entries, 0 to 1474
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Pays bénéficiaire  1475 non-null   object
 1   Année              1475 non-null   int64 
 2   Produit            1475 non-null   object
 3   Valeur             1475 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 46.2+ KB


In [36]:
#Je cherche les valeurs en doubles
df_aa.duplicated().sum()

0

In [37]:
#Je me renseigne sur les données statistiques des données.
df_aa.describe()

Année         Valeur
count  1475.000000    1475.000000
mean   2014.054237    7481.966780
std       0.946916   23339.531424
min    2013.000000       0.000000
25%    2013.000000     287.500000
50%    2014.000000    1178.000000
75%    2015.000000    4334.500000
max    2016.000000  265013.000000

In [38]:
#Ensuite je renomme les colonnes suivantes:
df_aa=df_aa.rename(columns={'Pays bénéficiaire':'Pays','Valeur':'Quantité'})
df_aa

Pays  Année                   Produit  Quantité
0     Afghanistan   2013       Autres non-céréales       682
1     Afghanistan   2014       Autres non-céréales       335
2     Afghanistan   2013              Blé et Farin     39224
3     Afghanistan   2014              Blé et Farin     15160
4     Afghanistan   2013                  Céréales     40504
...           ...    ...                       ...       ...
1470     Zimbabwe   2015  Mélanges et préparations        96
1471     Zimbabwe   2013              Non-céréales      5022
1472     Zimbabwe   2014              Non-céréales      2310
1473     Zimbabwe   2015              Non-céréales       306
1474     Zimbabwe   2013                Riz, total        64

[1475 rows x 4 columns]

In [39]:
#Conversion pour avoir les mêmes unités entre les differentes données : *1000 (conversion de tonnes en kg)
df_aa['Quantité']=df_aa['Quantité']*1000

In [40]:
#Afficher les lignes ordonnées par ordre décroissant en fonction de la quantité 
df_aa.sort_values(['Quantité'])

Pays  Année               Produit   Quantité
1368                Tadjikistan   2015          Non-céréales          0
1338                  Sri Lanka   2015          Sucre, total          0
1386                      Tchad   2013           Lait, total          0
1362                Tadjikistan   2015           Lait, total          0
219         Chine, continentale   2014      Huiles végétales          0
...                         ...    ...                   ...        ...
1294              Soudan du Sud   2014              Céréales  213730000
1297              Soudan du Sud   2014  Céréales Secondaires  213730000
1014  République arabe syrienne   2014              Céréales  225007000
362                    Éthiopie   2013              Céréales  256196000
363                    Éthiopie   2014              Céréales  265013000

[1475 rows x 4 columns]

# **Partie 2 : Analyse des données**

#  Lexique/Précisions

- 1. La disponibilité intérieure =Nourriture + semences + Pertes + Aliments pour animaux + Traitement+Autres Utilisations
- 2. La colonne nourriture correspond à la disponibilité pour l'alimentation humaine(disponibilié alimentaire).
- 3. La disponibilité intérieure =production - exportations + importation + variation des stocks
- 4. La colonne 'origine' precise si le produit est d'origine animale ou végétale

# DEMANDES DE MARC

# Qh1:<a name="C7"><font color='Black'>Proportion de la population en état de sous-nutrition en 2017</font></a>:

In [41]:
#Afficher la table sous_nutrition
df_sn

Pays  Année  nb_hab_sousnut
0     Afghanistan   2013       8600000.0
1     Afghanistan   2014       8800000.0
2     Afghanistan   2015       8900000.0
3     Afghanistan   2016       9700000.0
4     Afghanistan   2017      10500000.0
...           ...    ...             ...
1213     Zimbabwe   2014             NaN
1214     Zimbabwe   2015             NaN
1215     Zimbabwe   2016             NaN
1216     Zimbabwe   2017             NaN
1217     Zimbabwe   2018             NaN

[1218 rows x 3 columns]

In [42]:
#Afficher la table population
df_pop

Pays  Année      nb_hab
0     Afghanistan   2013  32269589.0
1     Afghanistan   2014  33370794.0
2     Afghanistan   2015  34413603.0
3     Afghanistan   2016  35383032.0
4     Afghanistan   2017  36296113.0
...           ...    ...         ...
1411     Zimbabwe   2014  13586707.0
1412     Zimbabwe   2015  13814629.0
1413     Zimbabwe   2016  14030331.0
1414     Zimbabwe   2017  14236595.0
1415     Zimbabwe   2018  14438802.0

[1416 rows x 3 columns]

In [43]:
# On definit la population en 2017:
pop17 = df_pop[df_pop['Année']==2017]
pop17

Pays  Année      nb_hab
4                                Afghanistan   2017  36296113.0
10                            Afrique du Sud   2017  57009756.0
16                                   Albanie   2017   2884169.0
22                                   Algérie   2017  41389189.0
28                                 Allemagne   2017  82658409.0
...                                      ...    ...         ...
1390  Venezuela (République bolivarienne du)   2017  29402484.0
1396                                Viet Nam   2017  94600648.0
1402                                   Yémen   2017  27834819.0
1408                                  Zambie   2017  16853599.0
1414                                Zimbabwe   2017  14236595.0

[236 rows x 3 columns]

In [44]:
# On définit le nombre de personnes en état de sous_nutrition en 2017:
sn17=df_sn[df_sn['Année']==2017]
sn17

Pays  Année  nb_hab_sousnut
4                                Afghanistan   2017      10500000.0
10                            Afrique du Sud   2017       3100000.0
16                                   Albanie   2017        100000.0
22                                   Algérie   2017       1300000.0
28                                 Allemagne   2017             NaN
...                                      ...    ...             ...
1192  Venezuela (République bolivarienne du)   2017       8000000.0
1198                                Viet Nam   2017       6500000.0
1204                                   Yémen   2017             NaN
1210                                  Zambie   2017             NaN
1216                                Zimbabwe   2017             NaN

[203 rows x 3 columns]

In [45]:
#Jointure entre les deux tables(pop17 et sn17):
pop_sn_2017=pd.merge(pop17[['Pays','nb_hab']],sn17[['Pays','nb_hab_sousnut']],on='Pays',how='inner')
pop_sn_2017

Pays      nb_hab  nb_hab_sousnut
0                               Afghanistan  36296113.0      10500000.0
1                            Afrique du Sud  57009756.0       3100000.0
2                                   Albanie   2884169.0        100000.0
3                                   Algérie  41389189.0       1300000.0
4                                 Allemagne  82658409.0             NaN
..                                      ...         ...             ...
198  Venezuela (République bolivarienne du)  29402484.0       8000000.0
199                                Viet Nam  94600648.0       6500000.0
200                                   Yémen  27834819.0             NaN
201                                  Zambie  16853599.0             NaN
202                                Zimbabwe  14236595.0             NaN

[203 rows x 3 columns]

In [46]:
#Suppréssion des valeurs manquantes
pop_sn_2017=pop_sn_2017.dropna()

In [47]:
#On affiche le résultat:
prop_sn_2017=100 * pop_sn_2017['nb_hab_sousnut'].sum()/pop_sn_2017['nb_hab'].sum()
pop_sn_2017
print('Le taux de sous-nutrition pour 2017 est de:', round(100 * pop_sn_2017['nb_hab_sousnut'].sum()/pop_sn_2017['nb_hab'].sum(),3),'%')

Le taux de sous-nutrition pour 2017 est de: 12.812 %


# Qh2:<a name="C8"><font color='Black'>Nombre théorique de personnes qui pourraient être nourries(à partir de la disponibilité alimentaire mondiale)</font></a>

In [48]:
#Afficher la table de disponibilié_alimentaire en 2017:
df_da

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    NaN   
1       Afghanistan        Agrumes, Autres  vegetale                    NaN   
2       Afghanistan  Aliments pour enfants  vegetale                    NaN   
3       Afghanistan                 Ananas  vegetale                    NaN   
4       Afghanistan                Bananes  vegetale                    NaN   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    NaN   
15601  Îles Salomon    Viande de Volailles   animale                    NaN   
15602  Îles Salomon          Viande, Autre   animale                    NaN   
15603  Îles Salomon                    Vin  vegetale                    NaN   
15604  Îles Salomon         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      NaN       5.0      4.712329      0.20        0.77   
1                      NaN       1.0      3.534247      0.01        0.02   
2                      NaN       1.0      0.164384      0.01        0.03   
3                      NaN       0.0      0.000000       NaN         NaN   
4                      NaN       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  NaN      45.0     12.876712      4.28        1.41   
15601                  NaN      11.0      9.150685      0.69        1.14   
15602                  NaN       0.0      0.164384       NaN        0.04   
15603                  NaN       0.0      0.191781       NaN         NaN   
15604                  NaN       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      NaN                      NaN   
1           41.0                      2.0                     40.0   
2            2.0                      NaN                      2.0   
3            0.0                      NaN                      0.0   
4           82.0                      NaN                     82.0   
...          ...                      ...                      ...   
15600        3.0                      NaN                      0.0   
15601        2.0                      NaN                      2.0   
15602        0.0                      NaN                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     NaN        53.0       NaN         NaN   
1            39.0     2.0         3.0       NaN         NaN   
2             2.0     NaN         NaN       NaN         NaN   
3             0.0     NaN         NaN       NaN         NaN   
4            82.0     NaN         NaN       NaN         NaN   
...           ...     ...         ...       ...         ...   
15600         3.0     NaN         2.0       NaN         NaN   
15601         2.0     NaN         0.0       NaN         NaN   
15602         0.0     NaN         NaN       NaN         NaN   
15603         0.0     NaN         NaN       NaN         NaN   
15604         0.0     NaN         0.0       NaN         NaN   

       Variation de stock  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN  
...                   ...  
15600                 NaN  
15601                 NaN  
15602                 NaN  
15603                 NaN  
15604                 NaN  

[15605 rows x 18 columns]

In [49]:
#Afficher la table de population en 2017:
pop17

Pays  Année      nb_hab
4                                Afghanistan   2017  36296113.0
10                            Afrique du Sud   2017  57009756.0
16                                   Albanie   2017   2884169.0
22                                   Algérie   2017  41389189.0
28                                 Allemagne   2017  82658409.0
...                                      ...    ...         ...
1390  Venezuela (République bolivarienne du)   2017  29402484.0
1396                                Viet Nam   2017  94600648.0
1402                                   Yémen   2017  27834819.0
1408                                  Zambie   2017  16853599.0
1414                                Zimbabwe   2017  14236595.0

[236 rows x 3 columns]

In [50]:
#Jointure entre la table disopnibité alimentaire et la table population
df_da_2017=pd.merge(df_da,pop17,how='left',on='Pays')
df_da_2017

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    NaN   
1       Afghanistan        Agrumes, Autres  vegetale                    NaN   
2       Afghanistan  Aliments pour enfants  vegetale                    NaN   
3       Afghanistan                 Ananas  vegetale                    NaN   
4       Afghanistan                Bananes  vegetale                    NaN   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    NaN   
15601  Îles Salomon    Viande de Volailles   animale                    NaN   
15602  Îles Salomon          Viande, Autre   animale                    NaN   
15603  Îles Salomon                    Vin  vegetale                    NaN   
15604  Îles Salomon         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      NaN       5.0      4.712329      0.20        0.77   
1                      NaN       1.0      3.534247      0.01        0.02   
2                      NaN       1.0      0.164384      0.01        0.03   
3                      NaN       0.0      0.000000       NaN         NaN   
4                      NaN       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  NaN      45.0     12.876712      4.28        1.41   
15601                  NaN      11.0      9.150685      0.69        1.14   
15602                  NaN       0.0      0.164384       NaN        0.04   
15603                  NaN       0.0      0.191781       NaN         NaN   
15604                  NaN       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      NaN                      NaN   
1           41.0                      2.0                     40.0   
2            2.0                      NaN                      2.0   
3            0.0                      NaN                      0.0   
4           82.0                      NaN                     82.0   
...          ...                      ...                      ...   
15600        3.0                      NaN                      0.0   
15601        2.0                      NaN                      2.0   
15602        0.0                      NaN                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     NaN        53.0       NaN         NaN   
1            39.0     2.0         3.0       NaN         NaN   
2             2.0     NaN         NaN       NaN         NaN   
3             0.0     NaN         NaN       NaN         NaN   
4            82.0     NaN         NaN       NaN         NaN   
...           ...     ...         ...       ...         ...   
15600         3.0     NaN         2.0       NaN         NaN   
15601         2.0     NaN         0.0       NaN         NaN   
15602         0.0     NaN         NaN       NaN         NaN   
15603         0.0     NaN         NaN       NaN         NaN   
15604         0.0     NaN         0.0       NaN         NaN   

       Variation de stock   Année      nb_hab  
0                     NaN  2017.0  36296113.0  
1                     NaN  2017.0  36296113.0  
2                     NaN  2017.0  36296113.0  
3                     NaN  2017.0  36296113.0  
4                     NaN  2017.0  36296113.0  
...                   ...     ...         ...  
15600                 NaN  2017.0    636039.0  
15601                 NaN  2017.0    636039.0  
15602                 NaN  2017.0    636039.0  
15603                 NaN  2017.0    636039.0  
15604                 NaN  

In [51]:
#Remplacer les NAN par 0
df_da_2017=df_da_2017.fillna(0)
df_da_2017

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    0.0   
1       Afghanistan        Agrumes, Autres  vegetale                    0.0   
2       Afghanistan  Aliments pour enfants  vegetale                    0.0   
3       Afghanistan                 Ananas  vegetale                    0.0   
4       Afghanistan                Bananes  vegetale                    0.0   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    0.0   
15601  Îles Salomon    Viande de Volailles   animale                    0.0   
15602  Îles Salomon          Viande, Autre   animale                    0.0   
15603  Îles Salomon                    Vin  vegetale                    0.0   
15604  Îles Salomon         Épices, Autres  vegetale                    0.0   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      0.0       5.0      4.712329      0.20        0.77   
1                      0.0       1.0      3.534247      0.01        0.02   
2                      0.0       1.0      0.164384      0.01        0.03   
3                      0.0       0.0      0.000000      0.00        0.00   
4                      0.0       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  0.0      45.0     12.876712      4.28        1.41   
15601                  0.0      11.0      9.150685      0.69        1.14   
15602                  0.0       0.0      0.164384      0.00        0.04   
15603                  0.0       0.0      0.191781      0.00        0.00   
15604                  0.0       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      0.0                      0.0   
1           41.0                      2.0                     40.0   
2            2.0                      0.0                      2.0   
3            0.0                      0.0                      0.0   
4           82.0                      0.0                     82.0   
...          ...                      ...                      ...   
15600        3.0                      0.0                      0.0   
15601        2.0                      0.0                      2.0   
15602        0.0                      0.0                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     0.0        53.0       0.0         0.0   
1            39.0     2.0         3.0       0.0         0.0   
2             2.0     0.0         0.0       0.0         0.0   
3             0.0     0.0         0.0       0.0         0.0   
4            82.0     0.0         0.0       0.0         0.0   
...           ...     ...         ...       ...         ...   
15600         3.0     0.0         2.0       0.0         0.0   
15601         2.0     0.0         0.0       0.0         0.0   
15602         0.0     0.0         0.0       0.0         0.0   
15603         0.0     0.0         0.0       0.0         0.0   
15604         0.0     0.0         0.0       0.0         0.0   

       Variation de stock   Année      nb_hab  
0                     0.0  2017.0  36296113.0  
1                     0.0  2017.0  36296113.0  
2                     0.0  2017.0  36296113.0  
3                     0.0  2017.0  36296113.0  
4                     0.0  2017.0  36296113.0  
...                   ...     ...         ...  
15600                 0.0  2017.0    636039.0  
15601                 0.0  2017.0    636039.0  
15602                 0.0  2017.0    636039.0  
15603                 0.0  2017.0    636039.0  
15604                 0.0  

In [52]:
#Je calcule le nombre de kcal par aliments et par Pays: kcal/personne/jour * population puis je fait la somme des kcal au niveau mondial
#On suppose qu'une personne est rassasiée avec 2400 calories par jour en moyenne
nb_pers_nourries=sum(df_da_2017['nb_hab']*df_da_2017['dispo_al'])/2400
nb_pers_nourries

8716243594.72125

In [53]:
print('Le nombre de personnes qui pourraient être nourries en 2017 est de :',(nb_pers_nourries))

Le nombre de personnes qui pourraient être nourries en 2017 est de : 8716243594.72125


In [54]:
print('le pourcentage de personnes qui pourraient etre nourries en 2017 est de :',round(100*nb_pers_nourries/pop17['nb_hab'].sum(),3),'%')

le pourcentage de personnes qui pourraient etre nourries en 2017 est de : 115.475 %


# Qh3:<a name="C9"><font color='Black'>Nombre théorique de personnes qui pourraient être nourries à partir de la disponibilité alimentaire des produits végéteaux</font></a>

In [55]:
#On utilise le tableau de disponibilité alimentaire avec la jointure crée à l'étape précédente.
#On filtre la colonne 'Origine'afin de ne concerver que les produits végétaux.
df_dispo_veg=df_da_2017[(df_da_2017['Origine']=='vegetale')]
df_dispo_veg

Pays                Produit   Origine  Aliments pour animaux  \
1       Afghanistan        Agrumes, Autres  vegetale                    0.0   
2       Afghanistan  Aliments pour enfants  vegetale                    0.0   
3       Afghanistan                 Ananas  vegetale                    0.0   
4       Afghanistan                Bananes  vegetale                    0.0   
6       Afghanistan                  Bière  vegetale                    0.0   
...             ...                    ...       ...                    ...   
15595  Îles Salomon                 Sésame  vegetale                    0.0   
15596  Îles Salomon                    Thé  vegetale                    0.0   
15597  Îles Salomon                Tomates  vegetale                    0.0   
15603  Îles Salomon                    Vin  vegetale                    0.0   
15604  Îles Salomon         Épices, Autres  vegetale                    0.0   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
1                      0.0       1.0      3.534247      0.01        0.02   
2                      0.0       1.0      0.164384      0.01        0.03   
3                      0.0       0.0      0.000000      0.00        0.00   
4                      0.0       4.0      7.397260      0.02        0.05   
6                      0.0       0.0      0.246575      0.00        0.00   
...                    ...       ...           ...       ...         ...   
15595                  0.0       0.0      0.027397      0.00        0.00   
15596                  0.0       0.0      0.465753      0.00        0.04   
15597                  0.0       0.0      0.191781      0.00        0.00   
15603                  0.0       0.0      0.191781      0.00        0.00   
15604                  0.0       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
1           41.0                      2.0                     40.0   
2            2.0                      0.0                      2.0   
3            0.0                      0.0                      0.0   
4           82.0                      0.0                     82.0   
6            3.0                      0.0                      3.0   
...          ...                      ...                      ...   
15595        0.0                      0.0                      0.0   
15596        0.0                      0.0                      0.0   
15597        0.0                      0.0                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
1            39.0     2.0         3.0       0.0         0.0   
2             2.0     0.0         0.0       0.0         0.0   
3             0.0     0.0         0.0       0.0         0.0   
4            82.0     0.0         0.0       0.0         0.0   
6             3.0     0.0         0.0       0.0         0.0   
...           ...     ...         ...       ...         ...   
15595         0.0     0.0         0.0       0.0         0.0   
15596         0.0     0.0         0.0       0.0         0.0   
15597         0.0     0.0         0.0       0.0         0.0   
15603         0.0     0.0         0.0       0.0         0.0   
15604         0.0     0.0         0.0       0.0         0.0   

       Variation de stock   Année      nb_hab  
1                     0.0  2017.0  36296113.0  
2                     0.0  2017.0  36296113.0  
3                     0.0  2017.0  36296113.0  
4                     0.0  2017.0  36296113.0  
6                     0.0  2017.0  36296113.0  
...                   ...     ...         ...  
15595                 0.0  2017.0    636039.0  
15596                 0.0  2017.0    636039.0  
15597                 0.0  2017.0    636039.0  
15603                 0.0  2017.0    636039.0  
15604                 0.0  

In [56]:
#Je calcule le nombre de kcal par aliments et par Pays: kcal/personne/jour(origine végétale) * population, puis je fait la somme des kcal au niveau mondial
#On suppose qu'une personne est rassasiée avec 2400 calories par jour en moyenne
nb_pers_da_veg=sum(df_dispo_veg['nb_hab']*df_dispo_veg['dispo_al'])/2400
print('Le nombre de personnes qui pourraient etre nourries en 2017 avec des produits végétaux est de:',nb_pers_da_veg)

Le nombre de personnes qui pourraient etre nourries en 2017 avec des produits végétaux est de: 7191985088.125417


In [57]:
print('le pourcentage des personnes qui pourraient etre nourries en 2017 avec des produits végétaux est de :',round(100*nb_pers_da_veg/pop17['nb_hab'].sum(),3),'%')

le pourcentage des personnes qui pourraient etre nourries en 2017 avec des produits végétaux est de : 95.282 %


# Q4:<a name="C10"><font color='Black'>L’utilisation de la disponibilité intérieure:La part qui est attribuée à l’alimentation animale, celle qui est perdue et celle qui est concrètement utilisée pour l'alimentation humaine</font></a>

In [58]:
#On utilise le tableau disponibilité alimentaire avec la jointure crée à l'étape précédente.
#On utilise la colonne 'disponibilité intérieur' afin de calculer la part de l'alimentation animale
#Disponibilité intérieure=Nourriture + semences + Pertes + Aliments pour animaux + Traitement+Autres Utilisations

In [59]:
#Calcul de la part de disponibilité intérieure attribuée à l'alimentation animale 
Part_animaux=df_da_2017['Aliments pour animaux'].sum()*100/df_da_2017['dispo_int'].sum()
Part_animaux
print('Part attribuée à l alimentation animale est de:',round(Part_animaux,3),'%')

Part attribuée à l alimentation animale est de: 13.242 %


In [60]:
#Calcul des pertes
pertes=df_da_2017['Pertes'].sum()*100/df_da_2017['dispo_int'].sum()
pertes
print('Les pertes est de:',round(pertes,3),'%')

Les pertes est de: 4.607 %


In [61]:
#Calcul de la part de la disponibilité intérieure attribuée à la nourriture
Part_nourriture=df_da_2017['Nourriture'].sum()*100/df_da_2017['dispo_int'].sum()
Part_nourriture
print('la part attribuée à la nourriture est de:',round(Part_nourriture,3),'%')

la part attribuée à la nourriture est de: 49.51 %


# DEMANDES DE MELANIE

# Qh1:<a name="C11"><font color='Black'>Pays pour lesquels la proportion de personnes sous-alimentées est la plus forte en 2017</font></a>

In [62]:
#On utilise le tableau de sous_nutrition avec la jointure crée à la premirère question de Marc
pop_sn_2017

Pays      nb_hab  nb_hab_sousnut
0                               Afghanistan  36296113.0      10500000.0
1                            Afrique du Sud  57009756.0       3100000.0
2                                   Albanie   2884169.0        100000.0
3                                   Algérie  41389189.0       1300000.0
6                                    Angola  29816766.0       5800000.0
..                                      ...         ...             ...
192                            Turkménistan   5757667.0        200000.0
195                                 Ukraine  44487709.0       1500000.0
197                                 Vanuatu    285510.0         10000.0
198  Venezuela (République bolivarienne du)  29402484.0       8000000.0
199                                Viet Nam  94600648.0       6500000.0

[104 rows x 3 columns]

In [63]:
#Je calcule la proportion de personnes sous alimentées pour chaque pays
pop_sn_2017['Proportion']=pop_sn_2017['nb_hab_sousnut']*100/pop_sn_2017['nb_hab']
pop_sn_2017['Proportion']

0      28.928718
1       5.437666
2       3.467203
3       3.140917
6      19.452143
         ...    
192     3.473629
195     3.371718
197     3.502504
198    27.208586
199     6.870989
Name: Proportion, Length: 104, dtype: float64

In [64]:
pop_sn_2017['Proportion'].fillna(0)

0      28.928718
1       5.437666
2       3.467203
3       3.140917
6      19.452143
         ...    
192     3.473629
195     3.371718
197     3.502504
198    27.208586
199     6.870989
Name: Proportion, Length: 104, dtype: float64

In [65]:
#J'affiche les lignes ordonnées par ordre décroissant en fonction de la Proportion
pop_sn_2017.sort_values(['Proportion'],ascending=False).head(10)

Pays      nb_hab  nb_hab_sousnut  \
78                                        Haïti  10982366.0       5300000.0   
157  République populaire démocratique de Corée  25429825.0      12000000.0   
108                                  Madagascar  25570512.0      10500000.0   
103                                     Libéria   4702226.0       1800000.0   
100                                     Lesotho   2091534.0        800000.0   
183                                       Tchad  15016753.0       5700000.0   
161                                      Rwanda  11980961.0       4200000.0   
121                                  Mozambique  28649018.0       9400000.0   
186                                 Timor-Leste   1243258.0        400000.0   
0                                   Afghanistan  36296113.0      10500000.0   

     Proportion  
78    48.259182  
157   47.188685  
108   41.062924  
103   38.279742  
100   38.249438  
183   37.957606  
161   35.055619  
121   32.810898  
186   32.173531  
0     28.928718

# Q2:<a name="C12"><font color='Black'>Pays qui ont le plus bénéficié d’aide depuis 2013</font></a>

In [66]:
#Afficher la table de disponibilité alimentaire en 2017:
df_aa

Pays  Année                   Produit  Quantité
0     Afghanistan   2013       Autres non-céréales    682000
1     Afghanistan   2014       Autres non-céréales    335000
2     Afghanistan   2013              Blé et Farin  39224000
3     Afghanistan   2014              Blé et Farin  15160000
4     Afghanistan   2013                  Céréales  40504000
...           ...    ...                       ...       ...
1470     Zimbabwe   2015  Mélanges et préparations     96000
1471     Zimbabwe   2013              Non-céréales   5022000
1472     Zimbabwe   2014              Non-céréales   2310000
1473     Zimbabwe   2015              Non-céréales    306000
1474     Zimbabwe   2013                Riz, total     64000

[1475 rows x 4 columns]

In [67]:
# On définit le nombre de personnes qui ont le plus bénéficié d’aide depuis 2013:
Aa13=df_aa[df_aa['Année']>=2013]
Aa13
#Regroupement par pays et par somme de quantités d'aide alimentaire à partir de 2013:
Aa13=Aa13[['Pays','Quantité']].groupby('Pays').sum('Quantité')
Aa13

Quantité
Pays                   
Afghanistan   185452000
Algérie        81114000
Angola          5014000
Bangladesh    348188000
Bhoutan         2666000
...                 ...
Zambie          3026000
Zimbabwe       62570000
Égypte          1122000
Équateur        1362000
Éthiopie     1381294000

[76 rows x 1 columns]

In [68]:
#J'affiche les lignes ordonnées par ordre décroissant en fonction de la Proportion
Aa13.sort_values(['Quantité'],ascending=False).head(10)

Quantité
Pays                                        
République arabe syrienne         1858943000
Éthiopie                          1381294000
Yémen                             1206484000
Soudan du Sud                      695248000
Soudan                             669784000
Kenya                              552836000
Bangladesh                         348188000
Somalie                            292678000
République démocratique du Congo   288502000
Niger                              276344000

# Q3:<a name="C13"><font color='Black'>Pays ayant le plus de disponibilité par habitant en 2017</font></a>

In [69]:
#Afficher la table de disponibilité alimentaire en 2017:
df_da

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    NaN   
1       Afghanistan        Agrumes, Autres  vegetale                    NaN   
2       Afghanistan  Aliments pour enfants  vegetale                    NaN   
3       Afghanistan                 Ananas  vegetale                    NaN   
4       Afghanistan                Bananes  vegetale                    NaN   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    NaN   
15601  Îles Salomon    Viande de Volailles   animale                    NaN   
15602  Îles Salomon          Viande, Autre   animale                    NaN   
15603  Îles Salomon                    Vin  vegetale                    NaN   
15604  Îles Salomon         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      NaN       5.0      4.712329      0.20        0.77   
1                      NaN       1.0      3.534247      0.01        0.02   
2                      NaN       1.0      0.164384      0.01        0.03   
3                      NaN       0.0      0.000000       NaN         NaN   
4                      NaN       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  NaN      45.0     12.876712      4.28        1.41   
15601                  NaN      11.0      9.150685      0.69        1.14   
15602                  NaN       0.0      0.164384       NaN        0.04   
15603                  NaN       0.0      0.191781       NaN         NaN   
15604                  NaN       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      NaN                      NaN   
1           41.0                      2.0                     40.0   
2            2.0                      NaN                      2.0   
3            0.0                      NaN                      0.0   
4           82.0                      NaN                     82.0   
...          ...                      ...                      ...   
15600        3.0                      NaN                      0.0   
15601        2.0                      NaN                      2.0   
15602        0.0                      NaN                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     NaN        53.0       NaN         NaN   
1            39.0     2.0         3.0       NaN         NaN   
2             2.0     NaN         NaN       NaN         NaN   
3             0.0     NaN         NaN       NaN         NaN   
4            82.0     NaN         NaN       NaN         NaN   
...           ...     ...         ...       ...         ...   
15600         3.0     NaN         2.0       NaN         NaN   
15601         2.0     NaN         0.0       NaN         NaN   
15602         0.0     NaN         NaN       NaN         NaN   
15603         0.0     NaN         NaN       NaN         NaN   
15604         0.0     NaN         0.0       NaN         NaN   

       Variation de stock  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN  
...                   ...  
15600                 NaN  
15601                 NaN  
15602                 NaN  
15603                 NaN  
15604                 NaN  

[15605 rows x 18 columns]

In [70]:
#On utilise le tableau disponibilité alimentaire avec la jointure crée à l'étape précédente(df_da_2017).
df_da_2017

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    0.0   
1       Afghanistan        Agrumes, Autres  vegetale                    0.0   
2       Afghanistan  Aliments pour enfants  vegetale                    0.0   
3       Afghanistan                 Ananas  vegetale                    0.0   
4       Afghanistan                Bananes  vegetale                    0.0   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    0.0   
15601  Îles Salomon    Viande de Volailles   animale                    0.0   
15602  Îles Salomon          Viande, Autre   animale                    0.0   
15603  Îles Salomon                    Vin  vegetale                    0.0   
15604  Îles Salomon         Épices, Autres  vegetale                    0.0   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      0.0       5.0      4.712329      0.20        0.77   
1                      0.0       1.0      3.534247      0.01        0.02   
2                      0.0       1.0      0.164384      0.01        0.03   
3                      0.0       0.0      0.000000      0.00        0.00   
4                      0.0       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  0.0      45.0     12.876712      4.28        1.41   
15601                  0.0      11.0      9.150685      0.69        1.14   
15602                  0.0       0.0      0.164384      0.00        0.04   
15603                  0.0       0.0      0.191781      0.00        0.00   
15604                  0.0       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      0.0                      0.0   
1           41.0                      2.0                     40.0   
2            2.0                      0.0                      2.0   
3            0.0                      0.0                      0.0   
4           82.0                      0.0                     82.0   
...          ...                      ...                      ...   
15600        3.0                      0.0                      0.0   
15601        2.0                      0.0                      2.0   
15602        0.0                      0.0                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     0.0        53.0       0.0         0.0   
1            39.0     2.0         3.0       0.0         0.0   
2             2.0     0.0         0.0       0.0         0.0   
3             0.0     0.0         0.0       0.0         0.0   
4            82.0     0.0         0.0       0.0         0.0   
...           ...     ...         ...       ...         ...   
15600         3.0     0.0         2.0       0.0         0.0   
15601         2.0     0.0         0.0       0.0         0.0   
15602         0.0     0.0         0.0       0.0         0.0   
15603         0.0     0.0         0.0       0.0         0.0   
15604         0.0     0.0         0.0       0.0         0.0   

       Variation de stock   Année      nb_hab  
0                     0.0  2017.0  36296113.0  
1                     0.0  2017.0  36296113.0  
2                     0.0  2017.0  36296113.0  
3                     0.0  2017.0  36296113.0  
4                     0.0  2017.0  36296113.0  
...                   ...     ...         ...  
15600                 0.0  2017.0    636039.0  
15601                 0.0  2017.0    636039.0  
15602                 0.0  2017.0    636039.0  
15603                 0.0  2017.0    636039.0  
15604                 0.0  

In [71]:
#Regroupement par Pays et somme de la disponibilité alimentaire par habiatant totale de chaque pays 
df_da_sum=df_da_2017[['Pays','dispo_al']].groupby('Pays').sum('dispo_al')
df_da_sum

dispo_al
Pays                           
Afghanistan              2087.0
Afrique du Sud           3020.0
Albanie                  3188.0
Algérie                  3293.0
Allemagne                3503.0
...                         ...
Émirats arabes unis      3275.0
Équateur                 2346.0
États-Unis d'Amérique    3682.0
Éthiopie                 2129.0
Îles Salomon             2383.0

[174 rows x 1 columns]

In [72]:
#J'affiche les lignes ordonnées par ordre décroissant en fonction de la disponibilité par habitant
df_da_sum.sort_values(['dispo_al'],ascending=False).head(10)

dispo_al
Pays                           
Autriche                 3770.0
Belgique                 3737.0
Turquie                  3708.0
États-Unis d'Amérique    3682.0
Israël                   3610.0
Irlande                  3602.0
Italie                   3578.0
Luxembourg               3540.0
Égypte                   3518.0
Allemagne                3503.0

# Q4:<a name="C14"><font color='Black'>Pays ayant le moins de disponibilité par habitant en 2017</font></a>

In [73]:
#J'affiche les lignes ordonnées par ordre décroissant en fonction de la disponibilité par habitant
df_da_sum.sort_values(['dispo_al'],ascending=True).head(10)

dispo_al
Pays                                                
République centrafricaine                     1879.0
Zambie                                        1924.0
Madagascar                                    2056.0
Afghanistan                                   2087.0
Haïti                                         2089.0
République populaire démocratique de Corée    2093.0
Tchad                                         2109.0
Zimbabwe                                      2113.0
Ouganda                                       2126.0
Timor-Leste                                   2129.0

# Q5:<a name="C15"><font color='Black'>Pays qui semblent être le plus en difficulté, au niveau alimentaire en 2017:</font></a>

In [74]:
#Afficher la table de la sous_nutrition en 2017:
sn17

Pays  Année  nb_hab_sousnut
4                                Afghanistan   2017      10500000.0
10                            Afrique du Sud   2017       3100000.0
16                                   Albanie   2017        100000.0
22                                   Algérie   2017       1300000.0
28                                 Allemagne   2017             NaN
...                                      ...    ...             ...
1192  Venezuela (République bolivarienne du)   2017       8000000.0
1198                                Viet Nam   2017       6500000.0
1204                                   Yémen   2017             NaN
1210                                  Zambie   2017             NaN
1216                                Zimbabwe   2017             NaN

[203 rows x 3 columns]

In [75]:
#Regroupement par pays et nombres habitants sous nourris en 2017
sn17_sum=sn17[['Pays','nb_hab_sousnut']].groupby('Pays').sum('nb_hab_sousnut')
sn17_sum

nb_hab_sousnut
Pays                                 
Afghanistan                10500000.0
Afrique du Sud              3100000.0
Albanie                      100000.0
Algérie                     1300000.0
Allemagne                         0.0
...                               ...
États-Unis d'Amérique             0.0
Éthiopie                   21100000.0
Îles Cook                         0.0
Îles Marshall                     0.0
Îles Salomon                  10000.0

[203 rows x 1 columns]

In [76]:
#afficher par ordre décroissant en fonction du nombre de popoulation en sous_nutrition:
sn17_sum.sort_values(['nb_hab_sousnut'],ascending=False).head(10)

nb_hab_sousnut
Pays                                                      
Inde                                           190100000.0
Pakistan                                        24800000.0
Indonésie                                       23600000.0
Nigéria                                         22800000.0
Bangladesh                                      21500000.0
Éthiopie                                        21100000.0
Philippines                                     15700000.0
République-Unie de Tanzanie                     13400000.0
République populaire démocratique de Corée      12000000.0
Kenya                                           11900000.0

# Demandes de Julien

# Q1:<a name="C16"><font color='Black'>Répartition des céréales entre l’alimentation humaine (colonne Nourriture) et l’alimentation pour animaux</font></a>

In [77]:
#Afficher la table de disponibilité alimentaire en 2017:
df_da

Pays                Produit   Origine  Aliments pour animaux  \
0       Afghanistan       Abats Comestible   animale                    NaN   
1       Afghanistan        Agrumes, Autres  vegetale                    NaN   
2       Afghanistan  Aliments pour enfants  vegetale                    NaN   
3       Afghanistan                 Ananas  vegetale                    NaN   
4       Afghanistan                Bananes  vegetale                    NaN   
...             ...                    ...       ...                    ...   
15600  Îles Salomon       Viande de Suides   animale                    NaN   
15601  Îles Salomon    Viande de Volailles   animale                    NaN   
15602  Îles Salomon          Viande, Autre   animale                    NaN   
15603  Îles Salomon                    Vin  vegetale                    NaN   
15604  Îles Salomon         Épices, Autres  vegetale                    NaN   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
0                      NaN       5.0      4.712329      0.20        0.77   
1                      NaN       1.0      3.534247      0.01        0.02   
2                      NaN       1.0      0.164384      0.01        0.03   
3                      NaN       0.0      0.000000       NaN         NaN   
4                      NaN       4.0      7.397260      0.02        0.05   
...                    ...       ...           ...       ...         ...   
15600                  NaN      45.0     12.876712      4.28        1.41   
15601                  NaN      11.0      9.150685      0.69        1.14   
15602                  NaN       0.0      0.164384       NaN        0.04   
15603                  NaN       0.0      0.191781       NaN         NaN   
15604                  NaN       4.0      1.315068      0.21        0.15   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
0           53.0                      NaN                      NaN   
1           41.0                      2.0                     40.0   
2            2.0                      NaN                      2.0   
3            0.0                      NaN                      0.0   
4           82.0                      NaN                     82.0   
...          ...                      ...                      ...   
15600        3.0                      NaN                      0.0   
15601        2.0                      NaN                      2.0   
15602        0.0                      NaN                      0.0   
15603        0.0                      0.0                      0.0   
15604        0.0                      0.0                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
0            53.0     NaN        53.0       NaN         NaN   
1            39.0     2.0         3.0       NaN         NaN   
2             2.0     NaN         NaN       NaN         NaN   
3             0.0     NaN         NaN       NaN         NaN   
4            82.0     NaN         NaN       NaN         NaN   
...           ...     ...         ...       ...         ...   
15600         3.0     NaN         2.0       NaN         NaN   
15601         2.0     NaN         0.0       NaN         NaN   
15602         0.0     NaN         NaN       NaN         NaN   
15603         0.0     NaN         NaN       NaN         NaN   
15604         0.0     NaN         0.0       NaN         NaN   

       Variation de stock  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN  
...                   ...  
15600                 NaN  
15601                 NaN  
15602                 NaN  
15603                 NaN  
15604                 NaN  

[15605 rows x 18 columns]

In [78]:
#Obtenir les valeurs uniques de la colonne'Produit'
df_da['Produit'].sort_values().unique()

array(['Abats Comestible', 'Agrumes, Autres', 'Alcool, non Comestible',
       'Aliments pour enfants', 'Ananas', 'Animaux Aquatiques Autre',
       'Arachides Decortiquees', 'Avoine', 'Bananes', 'Bananes plantains',
       'Beurre, Ghee', 'Bière', 'Blé', 'Boissons Alcooliques',
       'Boissons Fermentés', 'Café', 'Cephalopodes', 'Citrons & Limes',
       'Coco (Incl Coprah)', 'Crustacés', 'Crème', 'Céréales, Autres',
       'Dattes', 'Edulcorants Autres', 'Feve de Cacao', 'Fruits, Autres',
       'Girofles', 'Graines Colza/Moutarde', 'Graines de coton',
       'Graines de tournesol', 'Graisses Animales Crue', 'Haricots',
       'Huil Plantes Oleif Autr', 'Huile Graines de Coton',
       "Huile d'Arachide", "Huile d'Olive", 'Huile de Coco',
       'Huile de Colza&Moutarde', 'Huile de Germe de Maïs',
       'Huile de Palme', 'Huile de Palmistes', 'Huile de Soja',
       'Huile de Son de Riz', 'Huile de Sésame', 'Huile de Tournesol',
       'Huiles de Foie de Poisso', 'Huiles de Poisson

In [79]:
#Création d'un dataframe à partir de la disponibilité alimentaire filtrée sur les céréales seulement en 2017
liste_céréales =['Blé','Riz (Eq Blanchi)','Orge','Maïs','Seigle','Avoine','Millet','Sorgho','Autres']
df_céréales=df_da[df_da['Produit'].isin(liste_céréales)]
df_céréales

Pays           Produit   Origine  Aliments pour animaux  \
7       Afghanistan               Blé  vegetale                    NaN   
32      Afghanistan              Maïs  vegetale                  200.0   
34      Afghanistan            Millet  vegetale                    NaN   
40      Afghanistan              Orge  vegetale                  360.0   
47      Afghanistan  Riz (Eq Blanchi)  vegetale                    NaN   
...             ...               ...       ...                    ...   
15537  Îles Salomon               Blé  vegetale                    NaN   
15568  Îles Salomon              Maïs  vegetale                    NaN   
15575  Îles Salomon              Orge  vegetale                    NaN   
15591  Îles Salomon  Riz (Eq Blanchi)  vegetale                    0.0   
15593  Îles Salomon            Sorgho  vegetale                    0.0   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
7                      NaN    1369.0    438.986301      4.69       36.91   
32                     NaN      21.0      6.849315      0.30        0.56   
34                     NaN       3.0      1.095890      0.02        0.08   
40                     NaN      26.0      8.000000      0.24        0.79   
47                     NaN     141.0     37.863014      0.27        2.70   
...                    ...       ...           ...       ...         ...   
15537                  NaN     184.0     69.506849      1.00        5.19   
15568                  NaN       1.0      0.410959      0.01        0.03   
15575                  NaN       0.0      0.191781       NaN        0.01   
15591                 12.0     623.0    174.684932      1.36       10.90   
15593                  NaN       NaN           NaN       NaN         NaN   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
7         5992.0                      NaN                   1173.0   
32         313.0                      0.0                      1.0   
34          13.0                      NaN                      NaN   
40         524.0                      NaN                     10.0   
47         461.0                      NaN                    119.0   
...          ...                      ...                      ...   
15537       14.0                      0.0                     14.0   
15568        0.0                      NaN                      0.0   
15575        1.0                      NaN                      1.0   
15591       49.0                      0.0                     47.0   
15593        0.0                      NaN                      0.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
7          4895.0   775.0      5169.0     322.0         NaN   
32           76.0    31.0       312.0       5.0         NaN   
34           12.0     1.0        13.0       0.0         NaN   
40           89.0    52.0       514.0      22.0         NaN   
47          422.0    24.0       342.0      14.0         NaN   
...           ...     ...         ...       ...         ...   
15537        14.0     NaN         NaN       NaN         NaN   
15568         0.0     NaN         NaN       NaN         NaN   
15575         0.0     NaN         NaN       NaN         1.0   
15591        36.0     1.0         3.0       0.0         NaN   
15593         NaN     NaN         NaN       NaN         NaN   

       Variation de stock  
7                  -350.0  
32                    NaN  
34                    NaN  
40                    0.0  
47                    NaN  
...                   ...  
15537                 0.0  
15568                 NaN  
15575                 NaN  
15591                 0.0  
15593                 NaN  

[1325 rows x 18 columns]

In [80]:
#Proportion de céréales utilisées pour l'alimentation animale
céréales_animaux=df_céréales['Aliments pour animaux'].sum()*100 /df_céréales['dispo_int'].sum()
céréales_animaux
print(round(céréales_animaux,3),'% des céréales utilisées pour les animaux')

35.911 % des céréales utilisées pour les animaux


In [81]:
#Proportion de céréales utilisées pour la nourriture
céréales_nourritures=df_céréales['Nourriture'].sum()*100/df_céréales['dispo_int'].sum()
céréales_nourritures
print(round(céréales_nourritures,3),'% de céréales utilisées pour la nourriture')

43.021 % de céréales utilisées pour la nourriture


# Q2:<a name="C17"><font color='Black'> L'exportation du Manioc par rapport à sa production en Thaïlande</font></a>

In [82]:
#Afficher les données du Manioc en Thailande à partir de la table Disponibilité alimentaire
df_manioc_thai=df_da[(df_da['Produit']=='Manioc') & (df_da['Pays']=='Thaïlande')]
df_manioc_thai

Pays Produit   Origine  Aliments pour animaux  \
13809  Thaïlande  Manioc  vegetale                 1800.0   

       Autres Utilisations  dispo_al  dispo_al_qté  dispo_mg  dispo_prot  \
13809               2081.0      40.0     35.616438      0.05        0.14   

       dispo_int  Exportations - Quantité  Importations - Quantité  \
13809     6264.0                  25214.0                   1250.0   

       Nourriture  Pertes  Production  Semences  Traitement  \
13809       871.0  1511.0     30228.0       NaN         0.0   

       Variation de stock  
13809                 0.0

In [83]:
#Calcul du rapport de l'exportation du manioc en Thaïlande par rapport à sa production
Manioc=df_manioc_thai['Exportations - Quantité']/df_manioc_thai['Production']*100
Manioc
print('Proportion de manioc exporté par rapport à sa production en Thaïlande en 2017:', round(Manioc,2),'%')

Proportion de manioc exporté par rapport à sa production en Thaïlande en 2017: 13809    83.41
dtype: float64 %


In [84]:
#Afficher la prportion de population sous nourris en Thaïlande
df_sn17=pop_sn_2017[pop_sn_2017['Pays']=='Thaïlande']
df_sn17

Pays      nb_hab  nb_hab_sousnut  Proportion
185  Thaïlande  69209810.0       6200000.0    8.958268